File: Assets/scripts/enemyBehavor.js
#pragma strict
private var score:int=0;
function Start () {

}

function Update () {
	if(transform.position.y<-3.5){
		Destroy(this.gameObject);//自销毁
	}
}


function OnCollisionEnter2D( col:Collision2D){

	if(col.gameObject.name=="zidan(Clone)"){
	Destroy(col.gameObject);
	Destroy(this.gameObject);
	
	}
	
	var num:GameObject;
	num=gameObject.Find("score");
	num.GetComponent(scoreCount).score+=100;
}

File: Assets/scripts/Cs/EnemyBehavor.cs
using UnityEngine;
using System.Collections;

public class EnemyBehavor : MonoBehaviour {
    private int score = 0;
    private GameObject num;
	// Use this for initialization
	void Start () {
        num = GameObject.Find("score");
	}
	
	// Update is called once per frame
	void Update () {
        if (transform.position.y < -3.5) {
            Destroy(this.gameObject);
        }
	}
    void OnCollisionEnter2D(Collision2D col) { 
        if(col.gameObject.name=="zidan(Clone)"){
            Destroy(col.gameObject);
            Destroy(this.gameObject);
            num.GetComponent<scoreCount>().score += 100;
        }
        if(col.gameObject.name=="hero"){
            GameObject.Find("HeroLife").GetComponent<HeroLife>().heroLife -= 1;
            Destroy(this.gameObject); 
        }

    }

}
